aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3427,15 +3429,34 @@ static void linalg_lstsq_out_info(
   auto input_working_copy = copyBatchedColumnMajor(input);
 
   // now the actual call that computes the result in-place (apply_lstsq)
-  lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
-
+  if (driver == "gelss" && input.device() != at::kCPU) {
+    if (input.numel() == 0 || input.size(-2) == 0 || input.size(-1) == 0) {
+        auto output_shape = input.sizes().vec();
+        output_shape.back() = other.size(-1);
+        solution = at::zeros(output_shape, input.options());
```

Comment:
Is this necessary?

Reply:
Sorry I should've communicated these changes.

The tests will actually fail without this check because it'll generate tensors similar to this `torch.empty((0, 1))`. The `narrow` code will lead to

```
  File "/home/ksm/pytorch/test/test_linalg.py", line 316, in test_linalg_lstsq
    res = torch.linalg.lstsq(a, b, rcond=rcond, driver=driver)
RuntimeError: start (0) + length (1) exceeds dimension size (0).
``` 

Do we want to remove the edge case handling to simplify the logic, and communicate in the docs that this will occur?